aiida/restapi/resources.py

```diff
@@ -207,6 +208,130 @@ def get(self, id=None, page=None):  # pylint: disable=redefined-builtin,invalid-
+                'resource_type': self.__class__.__name__,
+                'data': data,
+            },
+        )
+
+    def post(self):
+        """
+        POST method to pass query help JSON.
+
+        This assumes the posted data is a complete and valid QueryBuilder queryhelp JSON object.
+        We could create a RequestParser that completely redefines all valid keys in the queryhelp,
+        but for now, it's much easier to simply pass-through the posted queryhelp and make sure
+        it can be handled by the built-in NodeTranslator.
+
+        This uses the NodeTranslator in order to best return Nodes according to the general AiiDA
+        REST API data format, while still allowing the return of other AiiDA entities.
+
+        :return: QueryBuilder result of AiiDA entities in "standard" REST API format.
+        """
+        # pylint: disable=protected-access
+        self.trans._query_help = json.loads(request.json) if isinstance(request.json, str) else request.json
+        assert isinstance(self.trans._query_help, dict), (
+            'POSTed data MUST be a valid QueryBuilder.queryhelp dictionary. '
+            f'Got instead (type: {type(self.trans._query_help)}): {self.trans._query_help}'
+        )
+        self.trans.__label__ = self.trans._result_type = self.trans._query_help['path'][-1]['tag']
+
+        # Handle empty list projections
+        number_projections = len(self.trans._query_help['project'])
+        counter = 0
+        skip_tags = []
+        for tag, projections in tuple(self.trans._query_help['project'].items()):
+            if projections == [{'*': {}}]:
+                self.trans._query_help['project'][tag] = self.trans._default
+            elif not projections:
+                counter += 1
+                skip_tags.append(tag)
+
+        if counter == number_projections:
+            # No projections have been specified in the queryhelp.
+            # To be true to the QueryBuilder response, the last entry in path
+            # is the only entry to be returned, all without edges/links.
+            self.trans._query_help['project'][self.trans.__label__] = self.trans._default
+
+        self.trans.init_qb()
+
+        data = {}
+        if self.trans.get_total_count():
+            if counter == number_projections:
+                # "Normal" REST API retrieval can be used.
+                data = self.trans.get_results()
+            else:
+                # Since the "normal" REST API retrieval relies on single-tag retrieval,
+                # we must instead be more creative with how we retrieve the results here.
+                # So we opt for a dictionary, with the tags being the keys.
+                for tag in self.trans._query_help['project']:
+                    if tag in skip_tags:
+                        continue
+                    self.trans.__label__ = tag
+                    data.update(self.trans.get_formatted_result(tag))
+
+        # Remove all 'full_type's
```

Comment:
The issue is the design of the REST API. For the `NodeTranslator` (used by the this endpoint), it will automatically include `full_type` during the data formatting.
As this has no meaning for anything that's not a Node (User, Group, etc.), I remove it. But maybe I should only remove it for non-Nodes. I think there was a reason, maybe that it doesn't correctly get the values needed to properly create the `full_type` value?

Reply:
I remember I had a lot of issues with this, as it's a curiosity of the REST API.

Reply:
This also warrants an issue, if there isn't one already.

Reply:
I've opted instead to only removing `full_type` if it's value is `None`.
The issue is I cannot determine whether the resource is a Node or not. This field is _only_ valid for Nodes. So this will be the compromise.
I've also ensured that the `full_type` will indeed be `None` (and not `"|"`) if neither `node_type` or `process_type` is available from the query.